Makefile

```diff
@@ -17,6 +17,19 @@ ifndef BUILD_ID
     BUILD_ID=local
 endif
 
+# Check that given variables are set and all have non-empty values,
+# die with an error otherwise.
+#
+# Params:
+#   1. Variable name(s) to test.
+#   2. (optional) Error message to print.
+check_defined = \
+    $(strip $(foreach 1,$1, \
+        $(call __check_defined,$1,$(strip $(value 2)))))
+__check_defined = \
+    $(if $(value $1),, \
+      $(error Undefined $1$(if $2, ($2))))
+
```

Comment:
This looks cryptic to me :) why don't we use something below?

```
.PHONY: check_defined
check_env:
	@if test "$(API_KEY)" = ""; then \
		echo "To trigger the CentOS CI release build you need to specify the CentOS CI API key."; \
		exit 1; \
	elif test "$(RELEASE_VERSION)" = ""; then \
		echo "You need to specify the version you want to release"; \
		exit 1; \
fi
```

And then add it to `ci_release` like we do for rhel check_env

Reply:
@budhrg then use same for `check_env` check purpose => https://github.com/minishift/minishift-centos-iso/blob/master/Makefile#L72-L86

Reply:
Done @praveenkumar